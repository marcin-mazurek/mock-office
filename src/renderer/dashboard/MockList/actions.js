export const LOAD = 'mocks/LOAD';
export const UNLOAD = 'mocks/UNLOAD';

export const load = id => ({
  type: LOAD,
  id
});

export const unload = id => ({
  type: UNLOAD,
  id
});
