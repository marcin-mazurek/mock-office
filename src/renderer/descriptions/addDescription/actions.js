export const INIT = 'addDescription/INIT';
export const ADD = 'descriptions/ADD';

export const init = (serverId, descriptions) => ({
  type: INIT,
  serverId,
  descriptions
});

export const add = (serverId, descriptionId, descriptionPayload, title = 'untitled', interval, reuse, quantity, delay, requirements, blocking) => ({
  type: ADD,
  serverId,
  id: descriptionId,
  descriptionPayload,
  title,
  interval,
  reuse,
  quantity,
  delay,
  requirements,
  blocking
});
