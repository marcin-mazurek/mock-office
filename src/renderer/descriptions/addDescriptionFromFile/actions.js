export const INIT = 'pickDescriptionFile/INIT';

export const init = (serverId, files) => ({
  type: INIT,
  serverId,
  files
});
