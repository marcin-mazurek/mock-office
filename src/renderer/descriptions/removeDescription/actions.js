export const INIT = 'remove/init';
export const REMOVE = 'descriptions/REMOVE';

export const init = (serverId, descriptionId) => ({
  type: INIT,
  descriptionId,
  serverId
});

export const remove = (serverId, descriptionId) => ({
  type: REMOVE,
  serverId,
  descriptionId
});
