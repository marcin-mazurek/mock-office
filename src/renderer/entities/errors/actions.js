export const ADD = 'errors/ADD';

export const add = reason => ({
  type: ADD,
  reason
});
