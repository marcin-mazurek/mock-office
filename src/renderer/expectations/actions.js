export const ADD = 'expectations/ADD';
export const LOAD = 'expectations/LOAD';
export const UNLOAD = 'expectations/UNLOAD';

export const add = (serverId, expectations) => ({
  type: ADD,
  serverId,
  expectations
});

export const load = (serverId, expectationId, instanceId, quantity) => ({
  type: LOAD,
  serverId,
  expectationId,
  instanceId,
  quantity
});

export const unload = (serverId, expectationId) => ({
  type: UNLOAD,
  serverId,
  expectationId
});
