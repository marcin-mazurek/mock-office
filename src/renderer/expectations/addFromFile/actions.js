export const FILE_PICK = 'addExpectationFromFile/FILE_PICK';

export const filePick = files => ({
  type: FILE_PICK,
  files
});
