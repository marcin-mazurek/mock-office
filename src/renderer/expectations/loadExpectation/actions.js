export const INIT = 'loadExpectation/INIT';

export const init = (serverId, expectationId) => ({
  type: INIT,
  serverId,
  expectationId
});
