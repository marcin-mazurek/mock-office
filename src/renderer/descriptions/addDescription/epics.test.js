import { Map, Set } from 'immutable';
import configureMockStore from 'redux-mock-store';
import 'rxjs';
import { createEpicMiddleware } from 'redux-observable';
import { init } from './actions';

const mockedElectron = {
  remote: {
    require() {
      return {
        default: {
          find() {
            return {
              addDescription() {
                return 'description id';
              }
            };
          }
        }
      };
    }
  }
};

jest.mock('electron', () => mockedElectron);
const addDescriptionEpic = require('./epics').default;

let store;
const epicMiddleware = createEpicMiddleware(addDescriptionEpic);
const mockStore = configureMockStore([epicMiddleware]);

describe('addDescriptionEpic', () => {
  test('success flow snapshot', () => {
    store = mockStore(new Map({
      servers: new Map({
        entities: new Map({
          'server id': {
            type: 'server type',
            name: 'server name',
            port: 3000,
            id: 'server id'
          }
        }),
        selected: 'server id',
        running: new Set()
      }),
      descriptions: new Map()
    }));

    store.dispatch(init('serverId', [
      {
        title: 'description title',
        requirements: {
          url: '/some-path'
        },
        descriptionPayload: 'descriptionPayload'
      }
    ]));
    expect(store.getActions()).toMatchSnapshot();
  });
});
