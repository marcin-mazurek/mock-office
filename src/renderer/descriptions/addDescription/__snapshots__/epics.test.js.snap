// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addDescriptionEpic success flow snapshot 1`] = `
Array [
  Object {
    "serverId": "serverId",
    "descriptions": Array [
      Object {
        "requirements": Object {
          "url": "/some-path",
        },
        "descriptionPayload": "descriptionPayload",
        "title": "description title",
      },
    ],
    "type": "addDescription/INIT",
  },
  Object {
    "blocking": undefined,
    "delay": undefined,
    "id": "description id",
    "interval": undefined,
    "quantity": undefined,
    "requirements": Object {
      "url": "/some-path",
    },
    "reuse": undefined,
    "serverId": "serverId",
    "descriptionPayload": "descriptionPayload",
    "title": "description title",
    "type": "descriptions/ADD",
  },
]
`;
